oeps/best-practices/oep-0067-bp-tools-and-technology.rst

```diff
@@ -0,0 +1,342 @@
+The problems we're trying to solve are:
+
+    * It is difficult to know what the current best practice for a given area.
+
+    * It is unclear how to go about the discussion and decision making process
+      to change a best practice.
+
+:doc:`OEP-11 <oep-0011-bp-FED-technology>` serves as a great example for the
+usefulness of this kind of OEP but limits itself to be relevant only to
+frontend technologies.  This OEP would supersede :doc:`OEP-11
+<oep-0011-bp-FED-technology>` and encompass not only frontend technolog but any
+technology that:
+
+    * Is considered a best practice for the Open edX Platform
+
+    * Is relevant to a large subset of the Open edX Platform
+
+Specification
+*************
+
+.. note:: We'll fill this out before the OEP lands but leaving mostly blank for initial feedback.
+
+General Technology Selection
+============================
+
+This technology is not specific to frontend or backend code.
+
+#. **Codecov should be used to measure code covered by tests**
+
+   **Rationale**: It is important to measure the amount of code covered by our
+   automated test suites. By striving for a high level of test coverage, we can
+   reduce the number of bugs that can only be found via manual testing. another
+   added benefit comes when we include coverage runs in the CI process, as
+   contributors are automatically reminded to include tests for any new code.
+
+   The Open edX community has standardized around `Codecov`_ as a measurement tool.
+
+#. **Use Github actions for CI Testing**
+
+   **Rationale**: As a healthy open source project we need good continouse integration
+   testing to prevent the introduction of obvious bugs into the platform code.  The Github Actions
+   tool should be used to perform Continuous Integration(CI) testing for all Open edX repositories
+   that perform any sort of software testing.
+
+Frontend Technology Selection
+=============================
+
+.. note:: 
+
+   The phrase “frontend” is used to mean any part of the platform that is shown to
+   users. This encompasses views rendered in Python on the server, interactive
+   interfaces written using JavaScript, and CSS styling.
+   
+#. **Use React**
+
+   **Rationale**: React must be used for building new UIs, as it is
+   widely adopted by the community and strikes a balance between
+   flexibility and feature richness.
+
+   When building in existing Django server-rendered pages, one can use the
+   react_render helper method to bridge to React. This bridge provides an
+   easy way to pass data into a React component from Django via props. For
+   more information about react_render see `ReactRender`_.
+
+#. **Use Redux**
+
+   For state management of complex
+   client-side interactions, Redux must be used. This library was chosen
+   because it sees strong use in the React community, but is also flexible
+   enough to be used in situations where a hybrid React/Backbone architecture
+   exists.
```

Comment:
I remember David Joy saying at one point that Redux wasn't worth the hassle and that MFE devs should feel free to simply not use it.

I'm not sure if I remember that right or whether that statement still applies, I'd need @arbrandes or @brian-smith-tcril 's input.

(I guess this is the sort of thing that will make this OEP useful :)

Reply:
This is a copy from the current frontend OEP.  I think any corrections should be made as follow-up ADRs and updates to the OEP.